mips64 elf2bin segfaulting

Found this while building a release
```
$ make TARGET=mips64-linux-muslsf V=1
...
../../../../tools/elf2bin /Users/bcook/projects/mettle/mettle/build/mips64-linux-muslsf/bin/sniffer /Users/bcook/projects/mettle/mettle/build/mips64-linux-muslsf/bin/sniffer.bin
data @ 0x10b4c9000, mapping @ 0x10b573000
make[5]: *** [install-exec-hook] Segmentation fault: 11
make[4]: *** [install-exec-am] Error 2
make[3]: *** [install-am] Error 2
make[2]: *** [install-recursive] Error 1
make[1]: *** [install-recursive] Error 1
make: *** [/Users/bcook/projects/mettle/mettle/build/mips64-linux-muslsf/bin/mettle.built] Error 2
```

--- a/util/elf2bin.c
+++ b/util/elf2bin.c
@@ -66,8 +66,8 @@
 			bin_info.dynamic_linker_info = LONG(phdr->p_vaddr); \
 		} \
 	} \
-	while (LONG(shdr->sh_type) != SHT_STRTAB) shdr++; \
-	while ((symb < symb_end) && strcmp((char *)((unsigned char *)ehdr + LONG(shdr->sh_offset) + LONG(symb->st_name)), ENTRYPOINT) != 0) symb++; \
+	while (INT(shdr->sh_type) != SHT_STRTAB) shdr++; \
+	while ((symb < symb_end) && strcmp((char *)((unsigned char *)ehdr + LONG(shdr->sh_offset) + INT(symb->st_name)), ENTRYPOINT) != 0) symb++; \
 	if (symb < symb_end) { \
 		bin_info.start_function = LONG(symb->st_value); \
 	}
@@ -172,7 +172,7 @@ int main(int argc, char **argv)
 		} else {
 			phdr = (Elf64_Phdr *)(data + bswap64(ehdr->e_phoff));
 			shdr = (Elf64_Shdr *)(data + bswap64(ehdr->e_shoff));
-			while (bswap64(shdr->sh_type) != SHT_SYMTAB) shdr++;
+			while (ntohl(shdr->sh_type) != SHT_SYMTAB) shdr++;
 			symb = (Elf64_Sym *)(data + bswap64(shdr->sh_offset));
 			symb_end = (Elf64_Sym *)((void *)symb + bswap64(shdr->sh_size));
 			MAP_BE64

